File: Offer2/0034/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func pathSum(root *TreeNode, target int) (res [][]int) {
	var path []int
	var dfs func(*TreeNode, int)
	dfs = func(node *TreeNode, sum int) {
		if node == nil {
			return
		}
		path = append(path, node.Val)
		if node.Left == nil && node.Right == nil && sum == node.Val {
			res = append(res, append([]int{}, path...))
			path = path[:len(path)-1]
			return
		}
		dfs(node.Left, sum-node.Val)
		dfs(node.Right, sum-node.Val)
		path = path[:len(path)-1]
	}
	dfs(root, target)
	return res
}

File: Offer2/0034/main.py
from typing import List


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def pathSum(self, root: TreeNode, target: int) -> List[List[int]]:
        res: List[List[int]] = []
        path: List[int] = []

        def dfs(node: TreeNode, sum: int):
            if node == None:
                return
            nonlocal path
            path.append(node.val)
            if node.left == None and node.right == None and sum == node.val:
                res.append(path.copy())
                path = path[:len(path) - 1]
                return
            dfs(node.left, sum - node.val)
            dfs(node.right, sum - node.val)
            path = path[:len(path) - 1]

        dfs(root, target)
        return res
